### PR TITLE
fix configure scripts so they fail if python3 is used

### LINKED ISSUE #20: AX_PYTHON_DEVEL macro fails to report Python 3 is not supported
This ticket replaces one of the issues reported in #15.

When python3 is detected, the `configure` script should gracefully fail informing the user python3 is not supported.

However, this does not happen (see https://gist.github.com/cdeil/39e9404e91954cb67270#file-gistfile1-txt-L276).

The `AX_PYTHON_DEVEL` call in `configure.ac` should be changed to `AX_PYTHON_DEVEL([<'3.0'])`
and `autoreconf` should be run. We need to coordinate this with the maintainers of the group library.

### PR DESCRIPTION
**Pull request by olaurino**
Resolve #20 

# Description
This PR fixes a configure issue by which python 3 was not detected when building the extern dependencies.

I applied the fix to our version of the group and stk library. These fixes should be made into the CIAO codebase as well, but that can happen at a later time, especially considering that we are currently discussing a possible CM change to these libraries.

# Testing
I tested this fix locally together with #159.

### CODE REVIEW

File: extern/grplib-4.7/configure [hunk 1]
```diff
@@ -13898,7 +13898,8 @@ $debug ||
 if test -n "$CONFIG_FILES"; then
 
 
-ac_cr=''
+ac_cr='
+'
```
Comment (olaurino) [inline extern/grplib-4.7/configure]: I just noticed git automatically removed a spurious CR (Carriage Return ^M) character from here. This should not have any effect on the scripts, but I'll double check.

Comment (olaurino) [inline extern/grplib-4.7/configure]: I removed this change from the commit, just to be safe.

**Comment by DougBurke** [comment]
@olaurino so, is the issue that checks like `[>'2.7' <'3.0']` weren't error-ing out for Python 3 users?

What happens if a user tries to use Python 2.6 to build? Do we still get an error (i.e. I'm slightly concerned about the removal of the `>2.7` check)?

**Comment by olaurino** [comment]
We do, because of this line:
https://github.com/sherpa/sherpa/pull/160/files#diff-28700aed04a9a4f1ab9ff8f6bcb2dbe2R15

In any case, I checked the documentation of the macro, and I can indeed use ==2.7 in order to pinpoint the version number. I'll make this change and test it. I don't think it makes too much of a difference, but it's safer.

**Comment by DougBurke** [comment]
Sounds sensible to be more-explicit here. It's unlikely that people will use an older Python, but just in case!

**Comment by olaurino** [comment]
I tried making the change and I now remember why I didn't do it that way to begin with. The more I work with this macro the less I like it. I have a patch ready to be submitted, but I still have to properly test it. Eventually, the macro itself should be changed, but I am holding off any deeper changes until we decide the future of these OTS.

**Comment by olaurino** [comment]
This seems to work. It's ugly, but it's the least extensive change I could come up with.

**Comment by DougBurke** [comment]
I would say looks good to me, but its `autoconf`, so that's not possible ;-)

I think this is good to be merged, unless you want @cdeil to check it as this is part of the fix for #15 ?

**Comment by cdeil** [comment]
My default Python on the command is Python 3.5 and both with master and this branch I see:
```
$ python setup.py install
Traceback (most recent call last):
  File "setup.py", line 50, in <module>
    from helpers.extensions import static_ext_modules
  File "/Users/deil/code/sherpa/helpers/__init__.py", line 21, in <module>
    from clean import clean
ImportError: No module named 'clean'
```
I think that's what most users and devs will see, i.e. they don't even get far enough in the build to see this change in the configure.
I don't have 2.6 available any more for testing, so I don't know if the build error reporting is improved with this PR.

As I've mentioned elsewhere already, I think maybe starting by making `python setup.py egg_info` work with Python 3 and then trying to port parts of Sherpa to Python 3 in small pieces (while always keeping Python 2 compat) would be nice (i.e. a realistic option to get going, for me and others to contribute). I don't know if this comment has any relevant on this PR. I guess no, since Python 3 is prevented before and after this change by this configure?

**Comment by DougBurke** [comment]
@cdeil - thanks for the feedback.

So, I would say this PR can be merged, but we have more to do to address
the original issue.

**Comment by olaurino** [comment]
That is why I didn't ask @cdeil to confirm this one, and I just tested it together with #159.

**Pull request merged**